components/service/glean/src/test/java/mozilla/components/service/glean/ping/PingMakerTest.kt

```diff
@@ -100,13 +119,46 @@ class PingMakerTest {
     fun `collect() must report an empty string when no data is stored`() {
         val engine1Data = JSONArray(listOf("1", "2", "3"))
         val engine2Data = JSONArray(listOf("a", "b", "c"))
-        val maker = PingMaker(StorageEngineManager(storageEngines = mapOf(
-            "engine1" to MockStorageEngine(engine1Data),
-            "engine2" to MockStorageEngine(engine2Data)
-        ), applicationContext = mockApplicationContext))
+        val maker = PingMaker(
+            StorageEngineManager(
+                storageEngines = mapOf(
+                    "engine1" to MockStorageEngine(engine1Data),
+                    "engine2" to MockStorageEngine(engine2Data)
+                ),
+                applicationContext = mockApplicationContext
+            ),
+            mockApplicationContext
+        )
 
         // Gather the data. We expect an empty string
         val data = maker.collect("noSuchData")
         assertNull("We expect an empty string", data)
     }
+
+    @Test
+    fun `seq number must be sequential`() {
+        // NOTE: Using a "real" ApplicationContext here so that it will have
+        // a working SharedPreferences implementation
+        var applicationContext = RuntimeEnvironment.application.applicationContext
+        val maker = PingMaker(
+            StorageEngineManager(
+                storageEngines = mapOf(
+                    "engine1" to MockStorageEngine(JSONObject())
+                ),
+                applicationContext = applicationContext
+            ),
+            applicationContext
+        )
+
+        // Collect two pings, and make sure their seq numbers are sequential
```

Comment:
We should expand this test case or create other tests cases to make sure that:

- sequence numbers start from 0;
- that we gracefully handle corrupted data (e.g. strings instead of numbers when unpersisting sequence numbers);
- that two different ping types have difference sequences, both starting from 0.